### PR TITLE
Cleanup unused cacheless PLEG code

### PR DESCRIPTION
**Pull request by tallclair**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

The generic PLEG checks a `cacheEnabled()` condition in multiple places, which checks whether the `podCache != nil`. However, GenericPLEG is only ever instantiated with a non-nil cache, so these checks are unnecessary.

- Kubelet pod cache instantied unconditionally: https://github.com/kubernetes/kubernetes/blob/64354890642adcfbbe94604bc9873d13928303d6/pkg/kubelet/kubelet.go#L624
  - (which never returns a non-nil value: https://github.com/kubernetes/kubernetes/blob/64354890642adcfbbe94604bc9873d13928303d6/pkg/kubelet/container/cache.go#L81-L84
- NewGenericPLEG calls:
  1. https://github.com/kubernetes/kubernetes/blob/64354890642adcfbbe94604bc9873d13928303d6/pkg/kubelet/kubelet.go#L743
  2. https://github.com/kubernetes/kubernetes/blob/64354890642adcfbbe94604bc9873d13928303d6/pkg/kubelet/kubelet.go#L761

#### Special notes for your reviewer:

I recommend reviewing with `Hide whitespace` enabled.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/sig node
/triage accepted

### CODE REVIEW

File: pkg/kubelet/pleg/generic.go [hunk 1]
```diff
@@ -402,10 +397,6 @@ func computeEvents(logger klog.Logger, oldPod, newPod *kubecontainer.Pod, cid *k
 	return generateEvents(logger, pid, cid.ID, oldState, newState)
 }
 
-func (g *GenericPLEG) cacheEnabled() bool {
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/pleg/generic.go:405]: we at least need to check for nil in constructor.

Comment (tallclair) [reply pkg/kubelet/pleg/generic.go:405]: Done. I just put a panic in to fail fast. Or do you think I need to propogate an error instead?

**Pull request merged**